virttest/qemu_monitor.py

```diff
@@ -340,8 +340,9 @@ def _log_command(self, cmd, debug=True, extra_str=""):
         :param extra_str: Extra string would be printed in log.
         """
         if self.debug_log or debug:
+            vm_name = "" if not hasattr(self.vm, "name") else self.vm.name
             logging.debug("(monitor %s.%s) Sending command '%s' %s",
-                          self.vm.name, self.name, cmd, extra_str)
+                          vm_name, self.name, cmd, extra_str)
```

Comment:
Could you please help me understand what the issue is like and how it happens?

@ldoktor do you know about that?

Reply:
The commit is from last year so I am telling from memory right now but we ended up with a lot of errors like "VM has no attribute 'name'" because the vms were unpickled from the 'env' file but during the unpickling process included unpickling monitor objects which in turn had the code in `_log_command` executed and referring to attributes of the original vm which were not yet unpickled at the time. This issue perform an extra check soving this unpickling recursion.

Reply:
I did some full testing without this fix and it seems that issue has been fixed in some other way. This fix originates from the autotest virt-test where it was needed to run on python3. I have also used it in avocado so far without any harm but since everything runs fine without it let's stick to minimal changes and close this pull request.

Reply:
Tested and we can go on without this patch.